homeassistant/components/hassio/backup.py

```diff
@@ -0,0 +1,170 @@
+"""Backup functionality for supervised installations."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from pathlib import Path
+from typing import Any
+
+from aiohasupervisor import backups as supervisor_backups
+
+from homeassistant.components.backup import (
+    BackupAgent,
+    BackupProgress,
+    BackupReaderWriter,
+    BackupUploadMetadata,
+    BaseBackup,
+    LocalBackupAgent,
+    NewBackup,
+)
+from homeassistant.core import HomeAssistant
+
+from .handler import get_supervisor_client
+
+
+async def async_get_backup_agents(
+    hass: HomeAssistant,
+    **kwargs: Any,
+) -> list[BackupAgent]:
+    """Return the hassio backup agents."""
+    return [SupervisorLocalBackupAgent(hass)]
+
+
+class SupervisorLocalBackupAgent(LocalBackupAgent):
+    """Local backup agent for supervised installations."""
+
+    name = "local"
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the backup agent."""
+        super().__init__()
+        self._hass = hass
+        self._backup_dir = Path("/backups")
```

Comment:
We need /backups to be readable from core and an update of the supervisor mount API including the mount point as seen from core